vitrivr-engine-core/src/main/kotlin/org/vitrivr/engine/core/features/metadata/source/exif/ExifMetadataExtractor.kt

```diff
@@ -0,0 +1,114 @@
+import org.vitrivr.engine.core.model.retrievable.attributes.SourceAttribute
+import org.vitrivr.engine.core.model.types.Value
+import org.vitrivr.engine.core.operators.Operator
+import org.vitrivr.engine.core.source.MediaType
+import org.vitrivr.engine.core.source.file.FileSource
+import java.text.DateFormat
+import java.text.SimpleDateFormat
+import java.util.*
+
+
+fun parseJson(jsonString: String): Map<String, String> {
+    val map = mutableMapOf<String, String>()
+    val regex = """"([^"]*)"\s*:\s*"([^"]*)"""".toRegex()
+
+    regex.findAll(jsonString).forEach { matchResult ->
+        val (key, value) = matchResult.destructured
+        map[key] = value
+    }
+
+    return map
+}
+
+fun convertDate(date: String): Date {
+    val format = SimpleDateFormat("yyyy:MM:dd HH:mm:ss")
+    return format.parse(date)
+}
+
+fun convertType(directory: Directory, tagType: Int, type: String): Value<*> {
+    return when(type) {
+        "STRING" -> Value.String(directory.getString(tagType))
+        "BOOLEAN" -> Value.Boolean(directory.getBoolean(tagType))
+        "BYTE" -> Value.Byte(directory.getObject(tagType) as Byte)
+        "SHORT" -> Value.Short(directory.getObject(tagType) as Short)
+        "INT" -> Value.Int(directory.getInt(tagType))
+        "LONG" -> Value.Long(directory.getLong(tagType))
+        "FLOAT" -> Value.Float(directory.getFloat(tagType))
+        "DOUBLE" -> Value.Double(directory.getDouble(tagType))
+        "DATETIME" -> Value.DateTime(convertDate(directory.getString(tagType)))
+        else -> throw(IllegalArgumentException("Type not supported"))
+    }
+}
+
+class ExifMetadataExtractor(
+    input: Operator<Retrievable>,
+    field: Schema.Field<ContentElement<*>, MapStructDescriptor>,
+    persisting: Boolean = true
+) : AbstractExtractor<ContentElement<*>, MapStructDescriptor>(input, field, persisting, bufferSize = 1) {
+
+    val logger: KLogger = KotlinLogging.logger {}
+
+    override fun matches(retrievable: Retrievable): Boolean =
+        retrievable.filteredAttribute(SourceAttribute::class.java)?.source is FileSource /* && retrievable.type is MediaType.IMAGE */
+
+
+    override fun extract(retrievable: Retrievable): List<MapStructDescriptor> {
+        var metadata = ImageMetadataReader.readMetadata((retrievable.filteredAttribute(SourceAttribute::class.java)?.source as FileSource).path.toFile())
+
+        var columnValues = mutableMapOf<String, Value<*>>()
+
+        for (directory in metadata.directories) {
+            for (tag in directory.tags) {
+                val tagname = tag.tagName.replace(Regex("[^a-zA-Z0-9]"), "")
+                val fullname = "${directory.name.replace(Regex("[^a-zA-Z0-9]"), "")}_${tagname}"
+
+                if (fullname == "ExifSubIFD_UserComment") {
+                    val userComment = tag.description
+                    if (userComment.isNotEmpty()) {
+                        val jsonMap = parseJson(userComment)
```

Comment:
While I do like the approach for custom metadata do be in the user comment field, I think it is not wise to just generally assume this is JSON. I'd suggest to at least sanity check whether it _looks_ like JSON.
And otherwise treat this as textual information.

Reply:
I have now changed it so that if ExifSubIFD_UserComment is explicitly configured as a parameter, it will just be treated as textual information.